docs/topics/multiplatform/multiplatform-connect-to-apis.md

```diff
@@ -1,380 +1,276 @@
-will be required to have a body:
-
-  ```kotlin
-  // Common
-  expect interface Mascot {
-      open fun display(): String
-  }
-  
-  class MascotImpl : Mascot {
-      // it's ok not to implement `display()`: all `actual`s are guaranteed to have a default implementation
-  }
-  
-  // Platform-specific
-  actual interface Mascot {
-      actual fun display(): String {
-          TODO()
-      }
-  }
-  ```
-
-During each platform compilation, the compiler ensures that every declaration marked with the `expect` keyword in the common
-or intermediate source set has the corresponding declarations marked with the `actual` keyword in all platform source sets.
-The IDE provides tools that help you create the missing actual declarations.
-
-If you have a platform-specific library that you want to use in shared code while providing your own implementation for
-another platform, you can provide a `typealias` to an existing class as the actual declaration:
-
 ```kotlin
-expect class AtomicRef<V>(value: V) {
-    fun get(): V
-    fun set(value: V)
-    fun getAndSet(value: V): V
-    fun compareAndSet(expect: V, update: V): Boolean
+// In the iosMain source set:
+class IOSPlatform : Platform
+
+actual val platformModule = module {
+    single<Platform> { IOSPlatform() }
 }
 ```
 
-```kotlin
-actual typealias AtomicRef<V> = java.util.concurrent.atomic.AtomicReference<V>
-```
+Here, Koin DSLs create modules that define components for injection. You declare a module in common code with
+the `expect` keyword and then provide a platform-specific implementation for each platform using the `actual` keyword.
+The framework takes care of selecting the correct implementation at runtime.
+
+When you use a DI framework, you inject all the dependencies through this framework. The same logic applies to handling
+platform dependencies. We recommend sticking to DI if you already have it in your project instead of using the expected
+and actual functions manually. This way, you avoid mixing two different ways of injecting dependencies.
+
+Note that you don't have to implement the common interface in Kotlin. You can do it in other languages, for example in
+Swift, in a different _platform module_. Then, you provide the implementation from the iOS platform module using the DI
+framework:
+
+![Using dependency injection framework](expect-di-framework.png){width=500}
+
+This approach only works if you put implementations in platform modules. It isn't very scalable because your Kotlin
+Multiplatform module can't be self-sufficient − you should implement the common interface in a different module.
+
+<!-- If you're interested in expanding this functionality to a shared module, please vote for this issue [TODO create or find an issue] and describe your use case. -->
+
+## What's next?
+
+For more examples and information on the expect/actual mechanism, see [Expected and actual declarations](multiplatform-expect-actual.md).
```

Comment:
```suggestion
For more examples and information on expected and actual declarations, see [Expected and actual declarations](multiplatform-expect-actual.md).
```

Reply:
OK, then shouldn't it be "expect**ed**/actual mechanism"?

Reply:
Not really, the key words in the mechanism are "expect" and "actual"

Reply:
Then shouldn't "expect" and "actual" be inline code? Like we discussed in the past?